pennylane/decomposition/resources.py

```diff
@@ -485,3 +501,18 @@ def _controlled_x_rep(  # pylint: disable=too-many-arguments
         num_work_wires=num_work_wires,
         work_wire_type=work_wire_type,
     )
+
+
+def resolve_work_wire_type(base_work_wires, base_work_wire_type, work_wires, work_wire_type):
```

Comment:
For best practices, functions should be defined above their first use.

Reply:
Moved